lib/orogen/gen/project.rb

```diff
@@ -497,6 +497,29 @@ def using_typekit(typekit)
 
             include OpaqueHandling
 
+            def get_pkg_config_deps(includeSelf = true)
```

Comment:
This has nothing to do with Project. Move the proxy code into its own class/module (e.g. CppProxies) and then call this class in #generation. Make the use optional (in particular, from the command line).

Moreover, there's a scheme to handle pkg-config:
 - the codegen class defines a #dependencies method returning a list of BuilDependency objects (see Deployment#dependencies for a good starting point for your use-case)
 - the templates use the Generation.cmake_* helpers (defined in base.rb)

Finally, do NOT puts in the orogen code. Ever. The output of orogen needs to stay as lean as possible so that warnings are harder to miss. If you need debug output, use OroGen.debug